kedro/framework/context/context.py

```diff
@@ -267,18 +268,18 @@ def _get_catalog(
         )
         conf_creds = self._get_config_credentials()
 
-        hook_manager = get_hook_manager()
-        catalog = hook_manager.hook.register_catalog(  # pylint: disable=no-member
-            catalog=conf_catalog,
-            credentials=conf_creds,
-            load_versions=load_versions,
-            save_version=save_version,
-        )
-        if not isinstance(catalog, DataCatalog):
+        try:
+            catalog = settings.DATA_CATALOG_CLASS.from_config(
+                catalog=conf_catalog,
+                credentials=conf_creds,
+                load_versions=load_versions,
+                save_version=save_version,
+            )
+        except (TypeError, AttributeError) as exc:
             raise KedroContextError(
                 f"Expected an instance of `DataCatalog`, "
-                f"got `{type(catalog).__name__}` instead."
-            )
+                f"got `{type(settings.DATA_CATALOG_CLASS)}` instead."
+            ) from exc
```

Comment:
So actually this is something that @idanov also pointed out and I tend to agree - that we should let `settings.py` handle the validation, rather than add another layer in `KedroContext`. We've set a bad precedent with `config_loader` but we can remediate it as it's not released yet. 

This is what we do for `SESSION_STORE_CLASS` already - the `KedroSession` accesses it straightforwardly with `settings.SESSION_STORE_CLASS` and doesn't wrap it in exception handling.

Reply:
Ah good shout! I didn't realise that. I'll change it and do the `config_loader` as well.